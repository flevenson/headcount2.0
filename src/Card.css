.card-div {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-width: 220px;
  margin: 8px auto;
  height: 270px;
  box-shadow: 0 0 5px 1px #0000006e;
  border-radius: 5px;
  background-color: #9AC5C5
}

.location {
  font-weight: 600;
  margin-bottom: 18px;
  font-size: 17px;
  color: #FFF;
}

.year-and-data-below {
  color: #FFF7B2;
}

.year-and-data-above {
  color: #009E9E;
}