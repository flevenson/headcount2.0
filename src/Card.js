import React from 'react';
import './Card.css'

const Card = ({stats, location}) => {
  let cardCounter = 0;
  const yearData = Object.keys(stats);
  const SchoolScore = yearData.map(point => {
    return <p className={(stats[point] < .5) ? 'year-and-data-red' : 'year-and-data-green'}
              key={cardCounter++}>
              {point} :  {stats[point]} </p>
  })


  return(
    <div className='card-div'>
      <h1 className='location'>{location}</h1>
      {SchoolScore}
    </div>
  )
}

export default Card