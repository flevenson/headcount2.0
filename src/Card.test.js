import React from 'react';
import { shallow, mount } from 'enzyme';

import Card from './Card';

describe('Card', () => {

  it('should match a snapshot', () => {
    const mockData = [{
      location: 'Turing',
      stats: {
        '2018' : 1
      }
    },
    {
      location: 'Turing',
      stats: {
        '2018' : 1
      }
    }
    ];
    

    const wrapper = shallow(<Card {...mockData[0]} handleCompare={jest.fn()}/> );
    const mockEvent = {
      target: {
        getAttribute: () => 'Turing'
       }
    }

     wrapper.simulate('click', mockEvent);
    expect(wrapper).toMatchSnapshot();
  });



  
});