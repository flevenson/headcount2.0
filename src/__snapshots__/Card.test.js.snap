// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    handleClick={[Function]}
    location="Turing"
    stats={
        Object {
            "2018": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="location"
          name="Turing"
>
          Turing
</h1>,
        Array [
          <p
            className="year-and-data-above"
            name="Turing"
>
            2018
             : 
            1
             
</p>,
        ],
      ],
      "className": "card-div",
      "name": "Turing",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Turing",
          "className": "location",
          "name": "Turing",
        },
        "ref": null,
        "rendered": "Turing",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2018",
            " : ",
            1,
            " ",
          ],
          "className": "year-and-data-above",
          "name": "Turing",
        },
        "ref": null,
        "rendered": Array [
          "2018",
          " : ",
          1,
          " ",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="location"
            name="Turing"
>
            Turing
</h1>,
          Array [
            <p
              className="year-and-data-above"
              name="Turing"
>
              2018
               : 
              1
               
</p>,
          ],
        ],
        "className": "card-div",
        "name": "Turing",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
            "className": "location",
            "name": "Turing",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2018",
              " : ",
              1,
              " ",
            ],
            "className": "year-and-data-above",
            "name": "Turing",
          },
          "ref": null,
          "rendered": Array [
            "2018",
            " : ",
            1,
            " ",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
